packages/pieces/community/formaloo/src/index.ts

```diff
@@ -0,0 +1,104 @@
+import { HttpMethod, createCustomApiCallAction, httpClient } from "@activepieces/pieces-common";
+import { createPiece, PieceAuth, PiecePropValueSchema, Property } from "@activepieces/pieces-framework";
+import { PieceCategory } from "@activepieces/shared";
+import FormData from 'form-data';
+
+const formalooAuth = PieceAuth.CustomAuth({
+  required: true,
+  description: `
+  1. Go to https://beta.formaloo.com/profile#api-credentials
+  2. Scroll down and you will find the API key and API Secret
+  3. Copy and paste below the keys
+  `,
+  props: {
+    api_key: Property.ShortText({
+      displayName: 'API Key',
+      description: 'Enter the base URL',
+      required: true,
+    }),
+    api_secret: PieceAuth.SecretText({
+      displayName: 'API Secret',
+      description: 'Enter the api secret',
+      required: true
+    })
+  },
+  validate: async ({ auth }) => {
+    if (auth) {
+      try {
+        const token: string = await authenticate(auth)
+        return !!token && token !== '' ? { valid: true } : { valid: false, error: "" }
+      } catch (e: any) {
+        return {
+          valid: false,
+          error: 'Credentials are invalid. ' + e?.message,
+        };
+      }
+    }
+
+    return {
+      valid: false,
+      error: 'Invalid Keys. Check and retry again.'
+    }
+  }
+})
+
+const authenticate = async (auth: PiecePropValueSchema<typeof formalooAuth>) => {
```

Comment:
I think this is the wrong type, Please use OAuth2 Authentication with grantType: client_credentials it should solve all your problems

Reply:
Ok, let me take a look. I knew it was different from other oauths, just not sure how. Thanks.